### DISCUSSION TITLE
Pods duplicated after restore

### DISCUSSION THREAD

**Discussion by VphDreamer** [vmware-tanzu/velero#5057] [Community support Q&A]
Hello, 
I am working in an Openshift environment, recently I did some backups and restores for an application, and I noticed that some pods 
are duplicated after restore, the pods are associated to a deployment, the deployments display `1 desired 1 available`, but when I list the pods I notice there are two instead of one for each deployment. Did anyone notice this before ?

**Comment by sseago** [comment]
I've seen this with DaemonSets before, but not with Deployments. I suspect what happened was that the restored pod was somehow not associated with the restored Deployment, but I'm not sure what may be different about your environment. With Velero installed via OADP, our default openshift plugins actually skip restore of pods with owner references to prevent this sort of thing (since in the past we've seen this happen with DaemonSets), and normally it's fine to just have a new pod created by the restored Deployment. However,  there are cases where we must restore the pod even with OADP -- if we're using restic for volume restore, for example, or if there are restore hooks -- in those cases we do restore the pod, and I haven't run across cases where the pod is duplicated on a deployment.

Does this always happen for all pods, or just certain deployments or certain restores?
What version of OpenShift are you on?
Are you using an upstream Velero release or velero via OADP? What version?

**Reply by VphDreamer** [reply]
@sseago Hello, Sorry for the late reply, but, I took some time to understand what's happening. I'm not really sure about my conclusion.
But, to answer to your questions, I'm using an **Openshift of version 4.8.34**,  **the OADP version is 1.0.3**. I tried to do some tests, and the thing I notice is when I restore, only the pods that are with no volume which are duplicated. So, I am assuming, perhaps what's happening is that velero (in some way) restores these pods **(with no owner reference)** while it has only to restore the ones with a volume, and in the other hand, there is the restored deployment that starts another pod.
I would like also to indicate that this happened to me when I am working with the default openshift plugin. I worked before with a custom plugin named cpdbr (it's an IBM plugin), I didn't have this problem at all.

**Reply by sseago** [reply]
@VphDreamer I'm thinking that you may want to open an issue under https://github.com/openshift/oadp-operator and we can debug it there. It may be that we are seeing an upstream Velero bug which will need to be addressed here, but it could also be an issue with the openshift plugin. When you open the issue, if you could provide resource yaml files for deployments and pods (with duplicates) -- if possible, before backup and after restore, but post-restore with the duplicates is most important. Also, `velero restore logs` and `velero restore describe` output (and same for backup).

**Reply by sseago** [reply]
Oh -- one more question. When you restore, are you restoring into an empty namespace, or is stuff already there? Specifically, are any of the pods or deployments from the backup already there?
